guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,954 @@
+   |
+   = note: required because of the requirements on the impl of `diesel::associations::HasTable` for `&diesel_demo::models::User`
+
+continued stack trace...
+```
+
+The first error shows that the compiler is expecting to find a field `id` on the struct
+even though we wanted the method.
+The second error explicitly states the `Identifiable` trait is not implemented for our `User` struct.
+
+The compiler is also giving us some more hints about trait bounds in each one of those *note* sections.
+The error messages regarding `IntoUpdateTarget` and `HasTable` are due to the trait bounds
+on the `update()` method that we called after `id()`.
+
+## AsChangeset
+
+As we've seen with inserting data, updating data with tuples can also become tedious.
+`AsChangeset` serves a similar purpose to `Insertable` in that
+it gives us a way to easily update a large amount of deserialized data.
+To derive the `AsChangeset` trait, add
+[`#[derive(AsChangeset)]`][aschangeset_doc] to your struct.
+
+In this section we will change our schema to make our `email` field nullable.
+`AsChangeset` allows us to deal with updating nullable fields a few different ways,
+which are changing the value, nulling the value, or ignoring the field.
+
+The nullable fields on our structs will now be of type `Option<T>`.
+Usually you do not want to change the primary key of the row or rows that you're updating.
+For this reason, we don't want to have `AsChangeset` and `Queryable` annotated on the same struct,
+which means we'll need another struct for updating our records.
+
+[aschangeset_doc]: https://docs.diesel.rs/diesel/query_builder/trait.AsChangeset.html
+
+Before we dive into some examples,
+let us take a look at our new schema.
+
+```rust
+// Output of "diesel print-schema"
+
+table! {
+    users (id) {
+        id -> Int4,
+        first_name -> Varchar,
+        last_name -> Varchar,
+        email -> Nullable<Varchar>,
+    }
+}
+```
+
+`AsChangeset` will automatically assume you don't want to change your record's primary key,
+so it will ignore that column.
+If your primary key field name is different than `id`,
+you must annotate your struct with the `#[primary_key(your_key)]` attribute.
+
+By default, `AsChangeset` will assume that anytime a field has the value `None`,
+we do not want to assign any values to it (ignore).
+If we truly want to assign a `NULL` value,
+we can use the annotation `#[changeset_options(treat_none_as_null="true")]`.
+Be careful, as when you are setting your `Option<T>` fields to `None`,
+they will be `NULL` in the database instead of ignored.
+
+However, there is a way to have both `AsChangeset` behaviors on a single struct.
+Instead of using the field type: `Option<T>`,
+you may use `Option<Option<T>>`.
+When updating, a value of `None` will be ignored and a value of `Some(None)` will `NULL` that column.
+All three options are shown in the following example code.
+Notice some of the code uses `unwrap_or()`.
```

Comment:
_Here,_ you can totally recycle that colon, though, and use it instead of the period